prometheus/map.jinja

```diff
@@ -1,16 +1,96 @@
-{% set server = salt['grains.filter_by']({
-    'default': {
-        'dir': {
-            'config': '/srv/volumes/prometheus',
-            'config_in_container': '/opt/prometheus/config'
-        },
-    },
-}, merge=salt['pillar.get']('prometheus:server')) %}
-
-{% set alertmanager = salt['grains.filter_by']({
-    'default': {
-        'dir': {
-            'config': '/srv/volumes/prometheus',
-        },
-    },
-}, merge=salt['pillar.get']('prometheus:alertmanager')) %}
+
+{%- set version = pillar.prometheus.get('server', {}).get('version', '1.6.2') %}
+
+{%- load_yaml as base_defaults %}
+{%- if pillar.docker is defined and pillar.docker.host is defined %}
+Debian:
+  dir:
+    config: /srv/volumes/prometheus
+    config_in_container: /opt/prometheus/config
+{%- else %}
+Debian:
+  dir:
+    config: /etc/prometheus
+    storage: /var/lib/prometheus/metrics
+    install: /opt
+    version_path: /opt/prometheus-{{ version }}.linux-amd64
+  version: {{ version }}.linux-amd64
+  source: https://github.com/prometheus/prometheus/releases/download/v{{ version }}/prometheus-{{ version }}.linux-amd64.tar.gz
```

Comment:
I would prefer to use "standarized" metadata for selecting installation method:
```
source:
  engine: tarball
  url: https://...
  hash: blahblah
```

in the similar manner as eg. for salt-formula-aptly, salt-formula-letsencrypt, etc.